Sources/WhisperKit/Core/TextDecoder.swift

```diff
@@ -478,12 +470,29 @@ open class TextDecoder: TextDecoding, WhisperMLModel {
         let samplingTime = Date().timeIntervalSince(samplingStartTime)
         timings.decodingSampling += samplingTime
 
-        let detectedLanguage = tokenizer.decode(tokens: [nextToken]).dropFirst(2).dropLast(2)
-        var decodingResult = DecodingResult.emptyResults
-        decodingResult.timings = timings
-        decodingResult.language = String(detectedLanguage)
-        Logging.debug("Detected language: \(detectedLanguage)")
-        return decodingResult
+        let sampledLanguage = String(tokenizer.decode(tokens: [nextToken]).dropFirst(2).dropLast(2))
+        let detectedLanguage: String
+        if Constants.languageCodes.contains(sampledLanguage) {
+            detectedLanguage = sampledLanguage
+            Logging.debug("Detected language: \(sampledLanguage)")
+        } else {
+            detectedLanguage = Constants.defaultLanguageCode
+            Logging.error("Detected language \(sampledLanguage) is not supported, defaulting to \(Constants.defaultLanguageCode)")
+        }
+        return DecodingResult(
+            language: detectedLanguage,
+            languageProbs: [:],
+            tokens: [],
+            tokenLogProbs: [],
+            text: "",
+            avgLogProb: 0.0,
+            noSpeechProb: 0.0,
+            temperature: 0.0,
+            compressionRatio: 0.0,
+            cache: nil,
+            timings: timings,
+            fallback: nil
+        )
```

Comment:
This seems reasonable, can it also return the `languageProbs` here?

Reply:
> This seems reasonable, can it also return the `languageProbs` here?

so detected language together with languageProbs instead of DecodingResult? will do!

Reply:
Might be better to keep an empty DecodingResult because it needs timings as well

Reply:
Just looked, I dont think the timings are being accounted for with this at all, would be nice to merge these in this PR too while you're in the code https://github.com/argmaxinc/WhisperKit/blob/c20943d3f5d6f64981e9eaed8a3a9c3a62124fc5/Sources/WhisperKit/Core/TranscribeTask.swift#L273-L319

Reply:
ok, done, please take look